feat: Add support for Protected Environments (#2084)
python-gitlab/python-gitlab: docs/gl_objects/protected_environments.rst

######################
Protected environments
######################

> You can define a list of protected environments on a repository.

Reviewer: ```suggestion
You can list and manage protected environments in a project.
```
